src/createInverseContext.tsx

```diff
@@ -0,0 +1,42 @@
+import React from "react";
+
+export const createInverseContext = <T extends unknown = null>(
+  initialValue: T,
+) => {
+  const Context = React.createContext(initialValue) as React.Context<T> & {
+    __updateProviderValue?: (value: T) => void;
```

Comment:
```suggestion
    _updateProviderValue?: (value: T) => void;
```
let's use a single underscore here ?